### PR TITLE
Clean up argument/option parsers

### LINKED ISSUE #43: Readability improvement for parsers in executable.py
The list of parsers in `executable.py` are difficult to read and could be improved with proper indentation.

I'll take a crack at this.

### PR DESCRIPTION
**Pull request by major**
This patch does a few things:

* Indents parser functions to make them more readable
* Groups each parser function along with its subcommand (with comments)
* Organizes subcommands alphabetically

Fixes #43.

Signed-off-by: Major Hayden <major@mhtx.net>

### CODE REVIEW

**Comment by spbnick** [comment]
Thank you, Major. Thanks for the review, @veruu!

I get this trying to run `skt --help`, though:

    Traceback (most recent call last):
      File "/home/nkondras/.local/bin/skt", line 11, in <module>
        load_entry_point('skt', 'console_scripts', 'skt')()
      File "/home/nkondras/projects/github.com/rh-fmk/skt/skt/executable.py", line 716, in main
        parser = setup_parser()
      File "/home/nkondras/projects/github.com/rh-fmk/skt/skt/executable.py", line 437, in setup_parser
        parents=[parser_merge, parser_build, parser_publish, parser_run,
    UnboundLocalError: local variable 'parser_merge' referenced before assignment

**Comment by major** [comment]
Weird. I'll go back and take another look.

File: skt/executable.py [hunk 1]
```diff
@@ -610,6 +751,13 @@ def main():
     global retcode
 
     parser = setup_parser()
+
+    # If the user ran 'skt' with no arguments or subcommands, print the help
+    # text rather than printing "error: too few arguments".
+    if len(sys.argv[1:])  == 0:
```
Comment (veruu) [inline skt/executable.py]: Besides having two spaces before the operator, pylint (which we don't run on the code yet) will complain about using `len()` in comparisons. Can we change this to `if not sys.argv[1:]:` instead?

Comment (major) [reply skt/executable.py]: @veruu Sure, good catch! :)

**Comment by major** [comment]
@veruu Should be fixed now.

**Comment by veruu** [comment]
LGTM now :+1:

File: skt/executable.py [hunk 2]
```diff
@@ -397,125 +397,266 @@ def setup_parser():
...
+    )
+    parser_merge.add_argument(
+        "-b", "--baserepo",
+        type=str,
+        help="Base repo URL"
+    )
+    parser_merge.add_argument(
+        "--ref",
+        type=str,
+        help="Base repo ref (default: master"
```
Comment (veruu) [inline skt/executable.py]: Argh, missed the missing ending paren!

File: skt/executable.py [hunk 3]
```diff
@@ -397,125 +397,266 @@ def setup_parser():
...
-                        "of rc file", action="store_true", default=False)
+    parser.add_argument(
+        "--state",
+        help="Save/read state from 'state' section of rc file",
+        action="store_true",
+        default=False
+    )
+    parser.add_argument(
+        "-h", "--help",
+        help="Merge sub-command help",
```
Comment (veruu) [inline skt/executable.py]: Hm, shouldn't this be a generic help? I never reads actual messages, unless I start debugging things...

**Comment by major** [comment]
Wait on merging this for now. I'm working through some issues coming from this PR.

**Pull request closed without merging**